yt_dlp/extractor/fikfap.py

```diff
@@ -0,0 +1,126 @@
+
+from .common import InfoExtractor
+from ..utils import OnDemandPagedList, unified_strdate
+
+
+class FikfapBaseIE(InfoExtractor):
+    _API_BASE = 'https://api.fikfap.com'
+
+    def _call_api(self, path, video_id, note='Downloading Post Information'):
+        return self._download_json(
+            f'{self._API_BASE}/{path}',
+            video_id,
+            headers={
+                'Origin': 'https://fikfap.com',
+                'Referer': 'https://fikfap.com/',
+                'Authorization-Anonymous': str(uuid.uuid4()),
+            },
+            note=note,
+        )
+
+    def _extract_post(self, post, video_id):
+        stream = post.get('videoStreamUrl')
+        if not stream:
+            return []
+        formats = self._extract_m3u8_formats(
+            stream, video_id,
+            ext='mp4',
+            headers={
+                'Origin': 'https://fikfap.com',
+                'Referer': 'https://fikfap.com/',
+            },
+        )
+        for fmt in formats:
+            fmt.setdefault('http_headers', {}).update({
+                'Origin': 'https://fikfap.com',
+                'Referer': 'https://fikfap.com/',
+            })
+
+        yield {
+            'id': video_id,
+            'formats': formats,
+            'title': post.get('label'),
+            'thumbnail': post.get('thumbnailStreamUrl'),
+            'likes_count': post.get('likesCount'),
+            'upload_date': unified_strdate(post.get('publishedAt')),
+            'view_count': post.get('viewsCount'),
+            'comments_count': post.get('commentsCount'),
+            'author': post.get('author'),
+            'age_limit': 18,
+        }
+
+
+class FikfapIE(FikfapBaseIE):
+    _VALID_URL = r'https?://(?:www\.)?fikfap\.com/(?:user/(?P<username>[^/]+)/post/|post/)(?P<id>\d+)'
+    _TESTS = [{
+        'url': 'https://fikfap.com/post/503729',
+        'info_dict': {
+            'id': '503729',
+        },
+    }]
+    IE_NAME = 'fikfap'
+
+    def _real_extract(self, url):
+        _, postid = self._match_valid_url(url).groups()
+        data = self._call_api(f'/posts/{postid}', postid)
+        return next(self._extract_post(data, postid))
+
+
+class FikfapPlaylistBaseIE(FikfapBaseIE):
+    def _real_extract(self, url):
+        video_id = self._match_valid_url(url).group('id')
+        amount = 21
+
+        def fetch_page(page):
+            after_id = None
+            if after_id is None:
+                path = f'/{self._MIDDLE_NAME}/{video_id}/posts?amount={amount}'
+            else:
+                path = f'/{self._MIDDLE_NAME}/{video_id}/posts?amount={amount}&afterId={after_id}'
+
+            data = self._call_api(path, video_id, f'Downloading Page {page}')
+            if not data:
+                return []
+
+            after_id = data[-1].get('postId')
+
+            entries = []
+            for pst in data:
+                entries.extend(self._extract_post(pst, video_id))
+
+            return entries
```

Comment:
why bother to use an `OnDemandPagedList` if you are just using an append/extend loop and returning `entries`??